frontend/src/v5/ui/routes/viewer/tickets/ticketsForm/properties/ticketImageList/ticketImageList.component.tsx

```diff
@@ -0,0 +1,214 @@
+			</IconText>
+		)}
+	</EmptyImageContainer>
+);
+
+const EnlargeImagesOverlay = ({ children, onClick }) => (
+	<OverlappingContainer onClick={onClick}>
+		{children}
+		<EnlargeContainer>
+			<EnlargeImageIcon />
+			<IconText>
+				<FormattedMessage id="viewer.cards.ticketImageList.enlarge" defaultMessage="Enlarge" />
+			</IconText>
+		</EnlargeContainer>
+	</OverlappingContainer>
+);
+
+export const TicketImageList = ({ value, onChange, onBlur, disabled, label, helperText, ...props }: FormInputProps) => {
+	const { isViewer } = useContext(TicketContext);
+	const getImgSrc = getImgSrcMapFunction();
+	const isProjectAdmin = ProjectsHooksSelectors.selectIsProjectAdmin();
+	const imgsSrcs = (value || []).map(getImgSrc);
+	const [imgsInModal, setImgsInModal] = useState(imgsSrcs);
+
+	const onClose = () => onChange(imgsInModal?.length ? imgsInModal.map(stripBase64Prefix) : null);
+	const onDeleteImages = () => onChange(null);
+	const onDeleteImage = (index) => setImgsInModal(imgsInModal.filter((img, i) => index !== i));
+	const onEditImage = (newValue, index) => {
+		if (newValue) {
+			imgsInModal[index] = newValue;
+			setImgsInModal([...imgsInModal]);
+		}
+	};
+
+	const uploadImages = async () => {
+		const files = await uploadFile(getSupportedImageExtensions(), true) as File[];
+		const imagesToUpload = [];
+		let corruptedImagesCount = 0;
+		let imagesTooBigCount = 0;
+		for (const file of files) {
+			if (imageIsTooBig(file)) {
+				imagesTooBigCount++;
+				continue;
+			}
+			try {
+				const imgSrc = await convertFileToImageSrc(file) as string;
+				await testImageExists(imgSrc);
+				imagesToUpload.push(imgSrc);
+			} catch (e) {
+				corruptedImagesCount++;
+			}
+		}
+		if (imagesToUpload.length) {
+			setImgsInModal(imgsInModal.concat(imagesToUpload));
+		}
+		if (imagesTooBigCount) {
+			DialogsActionsDispatchers.open('warning', {
+				title: formatMessage({
+					defaultMessage: 'Max file size exceeded',
+					id: 'ticketImageList.uploadImages.error.imageTooBig.title',
+				}),
+				message: formatMessage({
+					defaultMessage: `
+						{imagesTooBigCount} {imagesTooBigCount, plural, one {file was} other {files were}} too big and could not be uploaded.
+						The max file size is {maxFileSize}`,
+					id: 'ticketImageList.uploadImages.error.imageTooBig.message',
+				}, { imagesTooBigCount, maxFileSize: clientConfigService.resourceUploadSizeLimit }),
+			});
+		}
+		if (corruptedImagesCount) {
+			DialogsActionsDispatchers.open('warning', {
+				title: formatMessage({
+					defaultMessage: 'Invalid images',
+					id: 'ticketImageList.uploadImages.error.corruptedImage.title',
+				}),
+				message: formatMessage({
+					defaultMessage: '{corruptedImagesCount} {corruptedImagesCount, plural, one {file was} other {files were}} corrupted and could not be uploaded.',
+					id: 'ticketImageList.uploadImages.error.corruptedImage.message',
+				}, { corruptedImagesCount }),
+			});
+		}
```

Comment:
this (quite long) code is very similar to the one in currentUsercomponent. maybe we can extract the common parts to a helper file instead of duplicating it?